src/azul/indexer/document.py

```diff
@@ -665,25 +688,38 @@ def field_types(cls, field_types: FieldTypes) -> FieldTypes:
         return {
             **super().field_types(field_types),
             'num_contributions': pass_thru_int,
+            'sources': {
+                'id': pass_thru_str,
+                'name': pass_thru_str
+            },
             'bundles': {
                 'uuid': pass_thru_str,
                 'version': pass_thru_str,
             }
         }
 
     @classmethod
-    def _from_source(cls, source: JSON) -> Mapping[str, Any]:
-        return dict(super()._from_source(source),
-                    num_contributions=source['num_contributions'],
-                    bundles=source.get('bundles'))
+    def _from_json(cls, document: JSON) -> Mapping[str, Any]:
+        def entity_to_source_ref(entity):
+            return SourceRef(id=entity['id'], name=SimpleSourceName(entity['name']))
+
+        return dict(super()._from_json(document),
+                    num_contributions=document['num_contributions'],
+                    sources=map(entity_to_source_ref, document['sources']),
+                    bundles=document.get('bundles'))
 
     @classmethod
     def mandatory_source_fields(cls) -> List[str]:
-        return super().mandatory_source_fields() + ['num_contributions']
-
-    def to_source(self) -> JSON:
-        return dict(super().to_source(),
+        return super().mandatory_source_fields() + [
+            'num_contributions',
+            'sources.id',
+            'sources.name',
+        ]
+
+    def to_json(self) -> JSON:
+        return dict(super().to_json(),
                     num_contributions=self.num_contributions,
+                    sources=[source.to_json() for source in sorted(self.sources)],
```

Comment:
Can you motivate the sorting a litlle bit?

Reply:
Maybe sorting is unnecessary. I was just thinking that we would prefer a consistent document. But now I'm realizing that a consistent ordering is not really enforced at all for other fields, and that we work around the variable ordering of fields by sorting when we compare them during tests.

In conclusion, I don't have a strong justification here.

Reply:
When this was discussed, we decided that I should remove the sorting. That's what this last fixup commit does.